src/bin/console/console.cpp

```diff
@@ -495,33 +478,20 @@ int Console::visualize(const std::string& input) {
     mode = first_word;
   }
 
-  sql = input.substr(mode.size(), input.size());
   // Removes mode from sql string. If no mode is set, does nothing.
+  sql = input.substr(mode.size(), input.size());
 
-  SQLQueryPlan plan;
-  hsql::SQLParserResult parse_result;
-
-  try {
-    hsql::SQLParser::parse(sql, &parse_result);
-  } catch (const std::exception& exception) {
-    out("Exception thrown while parsing SQL query:\n  " + std::string(exception.what()) + "\n");
-    return ReturnCode::Error;
-  }
-
-  // Check if SQL query is valid
-  if (!parse_result.isValid()) {
-    out("Error: SQL query not valid.\n");
-    return ReturnCode::Error;
+  // If no SQL is provided, use the last execution. Else, create a new pipeline
+  if (!sql.empty()) {
+    _sql_pipeline = std::make_unique<SQLPipeline>(sql, _explicitly_created_transaction_context);
   }
 
   if (mode == "lqp" || mode == "lqpopt") {
     try {
-      auto lqp_roots = SQLTranslator{}.translate_parse_result(parse_result);
+      auto lqp_roots = _sql_pipeline->get_unoptimized_logical_plan();
```

Comment:
Same as in _eval_sql regarding the exceptions. Here we would basically double throw.

Reply:
Is fixed.